telethon/extensions/binaryreader.py

```diff
@@ -123,7 +123,7 @@ def tgread_date(self):
         if value == 0:
```

Comment:
We should remove this special case if we are considering any date as valid (like I said, this was for symmetry with the convenience of serializing `None` as empty timestamp or 0, but we don't *need* it).